api/apps/geoprocessing/src/export/pieces-exporters/project-config.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import { Readable } from 'stream';
+import { isLeft } from 'fp-ts/Either';
+
+import { ClonePiece, JobInput, JobOutput } from '@marxan/cloning';
+import { FileRepository } from '@marxan/files-repository';
+
+import { geoprocessingConnections } from '@marxan-geoprocessing/ormconfig';
+
+import { PieceExportProvider, PieceProcessor } from '../pieces/piece-processor';
+import { ResourceKind } from '@marxan/cloning/domain';
+
+@Injectable()
+@PieceExportProvider()
+export class ProjectConfig extends PieceProcessor {
+  constructor(
+    private readonly fileRepository: FileRepository,
+    @InjectEntityManager(geoprocessingConnections.apiDB)
+    private readonly entityManager: EntityManager,
+  ) {
+    super();
+  }
+
+  isSupported(piece: ClonePiece): boolean {
+    return piece === ClonePiece.ExportConfig;
+  }
+
+  async run(input: JobInput): Promise<JobOutput> {
+    if (input.resourceKind === ResourceKind.Scenario) {
+      throw new Error(`Exporting scenario is not yet supported.`);
+    }
+
+    await delay();
+
+    const scenarios: { name: string }[] = await this.entityManager.query(
+      `
+       SELECT name FROM scenarios where project_id = $1
+    `,
+      [input.resourceId],
+    );
+
+    const metadata = JSON.stringify({
```

Comment:
is this meant to mock the project metadata coming from the db?

Reply:
it doesn't mock, it should actually be a valid implementation of what's needed for config, according to initial assumptions/HLD.